Add plausible analytics (#828)
pydata/pydata-sphinx-theme: docs/user_guide/analytics.rst

Analytics and usage services
============================

> The theme supports analytics via the ``analytics`` option. It is configured

Reviewer: ```suggestion
The theme supports several web analytics services via the ``analytics`` option. It is configured
```

by passing a dictionary with options. See the sections bellow for relevant

> options depending on the analytic provider that you want to use.

Reviewer: ```suggestion
options depending on the analytics provider that you want to use.
```

.. code:: python

   html_theme_options = {
       "analytics": analytics_options,
   }

> Plausible Analytics (recommended)

Reviewer: I feel like we shouldn't say "recommended" in the title, given that it is not free for people who can't get a server via the scientific-python ecosystem, and people might use this theme that are not part of that community. Instead, what if we put an admonition like this at the top:

````
Generally speaking we recommend using Plausible over Google Analytics because it has a better story around user security and privacy.
In addition, it is more open-source and transparent.
In fact, `you can self-host a Plausible server <https://www.elvisduru.com/blog/how-to-self-host-plausible-analytics>`__.

.. admonition:: Get a self-hosted Plausible server at ``scientific-python.org``
   :class: tip

   If your documentation is for a package that is part of the SciPy / PyData ecosystem, they might be able to host a Plausible server for you at `<your-package>.scientific-python.org`.
   To ask about this, contact them on the social media platform of your choice and learn more at `scientific-python.org <https://scientific-python.org>`__.
````

Author: I will use your suggestion.

=================================

https://plausible.io can be used to gather simple
and privacy-friendly analytics for the site. The configuration consists in
a server URL and a specific domain. Plausible' javascript will be included in
all html pages to gather metrics. And the dashboard can be accessed at
``https://url/my_domain``.

> The Scientific-Python community can offer a self-hosted server. Contact the

Reviewer: Is this documented somewhere? We were looking into Plausible for Jupyter a while back, but didn't do it in-part because it was pretty expensive to run at the scale of Jupyter's website. I wonder if we could piggy-back on scientific-python with this?

Author: Documented as what is the process to get access? Nothing yet, this would be the first document talking about that. At some point we might have some mention on our website when we list all the initiatives we have.

@stefanv maybe you can answer this one? I don't know how hard this is for the server.
But I would guess it's ok. SciPy is getting 100k per month which is nothing. I don't know how many RPS the server can handle but I would assume >100 RPS (based on my personal experience, we should see this with e.g. locust to be sure)

team on social media following https://scientific-python.org for assistance.

.. code:: python

>    analytics_options = {

Reviewer: ```suggestion
   # To be re-used in html_theme_options["analytics"]
   analytics_options = {
```

Analytics and usage services
============================

The theme supports several web analytics services via the ``analytics`` option. It is configured
by passing a dictionary with options. See the sections bellow for relevant
options depending on the analytics provider that you want to use.

.. code:: python

   html_theme_options = {

>        "analytics": analytics_options,

Reviewer: ```suggestion
       # See below for options for each service
       "analytics": analytics_options,
```

   }

Generally speaking we recommend using Plausible over Google Analytics because
it has a better story around user security and privacy. In addition, it is more
open-source and transparent. In fact,
`you can self-host a Plausible server <https://www.elvisduru.com/blog/how-to-self-host-plausible-analytics>`__.

.. admonition:: Get a self-hosted Plausible server at ``scientific-python.org``
   :class: tip

   If your documentation is for a package that is part of the SciPy / PyData
   ecosystem, they might be able to host a Plausible server for you at

>    `<your-package>.scientific-python.org`.

Reviewer: ```suggestion
   ``<your-package>.scientific-python.org``.
```

   To ask about this, contact them on the social media platform of your choice
   and learn more at `scientific-python.org <https://scientific-python.org>`__.

Plausible Analytics
===================

> https://plausible.io can be used to gather simple

Reviewer: ```suggestion
`plausible.io <https://plausible.io>`__ can be used to gather simple
```

> and privacy-friendly analytics for the site. The configuration consists in
> a server URL and a specific domain. Plausible' javascript will be included in
> all html pages to gather metrics. And the dashboard can be accessed at
> ``https://url/my_domain``.

Reviewer: ```suggestion
and privacy-friendly analytics for the site. To configure, you will need to provide two things:

- A URL pointing to the JavaScript analytics script that is served by your Plausible server
- A domain that reflects where your documentation lives

Plausible' javascript will be included in all html pages to gather metrics.
The dashboard with analytics results will be accessible at ``https://<plausible-url>/<my-domain>``.
```

.. code:: python

   # To be re-used in html_theme_options["analytics"]
   analytics_options = {

>        "plausible_analytics_domain": "my-domain",
>        "plausible_analytics_url": "https://.../script.js",

Reviewer: ```suggestion
       # The domain you'd like to use for this analytics instance
       "plausible_analytics_domain": "my-domain",
       # The analytics script that is served by Plausible
       "plausible_analytics_url": "https://.../script.js",
```

   }

> 
> Google Analytics
> ================

Reviewer: ```suggestion

.. seealso::

  See the ``Plausible Documentation <https://plausible.io/docs/plausible-script>`__ for more information about this script.

Google Analytics
================
```

Analytics and usage services
============================

The theme supports several web analytics services via the ``analytics`` option. It is configured
by passing a dictionary with options. See the sections bellow for relevant
options depending on the analytics provider that you want to use.

.. code:: python

   html_theme_options = {
       # See below for options for each service
       "analytics": analytics_options,
   }

Generally speaking we recommend using Plausible over Google Analytics because
it has a better story around user security and privacy. In addition, it is more
open-source and transparent. In fact,
`you can self-host a Plausible server <https://www.elvisduru.com/blog/how-to-self-host-plausible-analytics>`__.

.. admonition:: Get a self-hosted Plausible server at ``scientific-python.org``
   :class: tip

   If your documentation is for a package that is part of the SciPy / PyData
   ecosystem, they might be able to host a Plausible server for you at

>    ``<your-package>.scientific-python.org``.

Reviewer 2: @tupui Note that we host all projects at `views.scientific-python.org`

Reviewer: Is there anything project-specific in a URL? Or just one big dashboard for everybody?

Author: Ah yes it should be `https://views.scientific-python.org/<your-package>`

   To ask about this, contact them on the social media platform of your choice
   and learn more at `scientific-python.org <https://scientific-python.org>`__.

Plausible Analytics
===================

`plausible.io <https://plausible.io>`__ can be used to gather simple
and privacy-friendly analytics for the site. To configure, you will need to provide two things:

- A URL pointing to the JavaScript analytics script that is served by your Plausible server
- A domain that reflects where your documentation lives

> Plausible' javascript will be included in all html pages to gather metrics.

Reviewer 2: Incorrect usage of apostrophe

The dashboard with analytics results will be accessible at ``https://<plausible-url>/<my-domain>``.

.. code:: python

   # To be re-used in html_theme_options["analytics"]
   analytics_options = {
       # The domain you'd like to use for this analytics instance
       "plausible_analytics_domain": "my-domain",
       # The analytics script that is served by Plausible
       "plausible_analytics_url": "https://.../script.js",
   }

.. seealso::

>   See the ``Plausible Documentation <https://plausible.io/docs/plausible-script>`__ for more information about this script.

Reviewer: ```suggestion
  See the `Plausible Documentation <https://plausible.io/docs/plausible-script>`__ for more information about this script.
```

   html_theme_options = {
       "google_analytics_id": "G-XXXXXXXXXX",
   }

Plausible Analytics
===================

Alternatively https://plausible.io can be used to gather simple
and privacy-friendly analytics for the site. The dashboard can be accessed

> at ``url/domain``.

Reviewer 2: I find `url/domain` a bit confusing here.

> The Scientific-Python community can offer a self-hosted server.

Reviewer 2: Provide instructions on how to request one?

> Plausible' javascript is included in the html pages.

Reviewer 2: "The Plausible Javascript needs to be included"? Or what is the intent here?

.. code:: python

>    analytics_options = {
>        "google_analytics_id": "G-XXXXXXXXXX",

Reviewer: ```suggestion
   # To be re-used in html_theme_options["analytics"]
   analytics_options = {
       "google_analytics_id": "G-XXXXXXXXXX",
```
